Assets/Scripts/WorldProperties.cs

```diff
@@ -91,6 +93,111 @@ void Start()
             clipShader = GameObject.FindWithTag("Ground").GetComponent<Renderer>().material.shader;
         }
 
+        /// <summary>
+        /// Cycle through the connected drones
+        /// </summary>
+        public static void SelectNextDrone()
+        {
+            Debug.Log("Selection next drone");
+
+            if(selectedDrone != null)
+            {
+                dronesQueue.Enqueue(selectedDrone);
+            }
+
+            if (dronesQueue.Count > 0)
+            {
+                Drone nextDrone = dronesQueue.Dequeue();
+                nextDrone.gameObjectPointer.GetComponent<DroneProperties>().SelectDrone();
```

Comment:
if you take my previous comment, here is where you would enqueue the drone back.